### DISCUSSION TITLE
Sharphook not working in published version of app

### DISCUSSION THREAD

**Discussion by BlackCoyote** [TolikPylypchuk/SharpHook#89] [Q&A] [answered]
Hey,

We're trying to build an Avalonia app that uses Sharphook, but for some reason we're getting some unexpected behaviour with Sharphook in the published version of the app.

Everything works as expected in the normal debug and release builds of the app, but once we try the publish output, any call to `EventSimulator.SimulateKeyPress()` will not work. It will throw the following exception for us:
```
System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (0x8007000B)
   at SharpHook.Native.UioHook.PostEvent(UioHookEvent& e)
   at SharpHook.Providers.UioHookProvider.PostEvent(UioHookEvent& e)
   at SharpHook.EventSimulator.SimulateKeyPress(KeyCode keyCode)
   at A.f.A.A()
   ```

When we tried using the [logging](https://sharphook.tolik.io/articles/logging.html#using-high-level-types) feature of sharphook, we experience a similar exception:
```
System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (0x8007000B)
   at SharpHook.Native.UioHook.SetLoggerProc(LoggerProc loggerProc, IntPtr userData)
   at SharpHook.Providers.UioHookProvider.SetLoggerProc(LoggerProc loggerProc, IntPtr userData)
   at SharpHook.Logging.LogSource.RegisterOrGet(ILoggingProvider loggingProvider, LogLevel minLevel)
   at SharpHook.Logging.LogSource.RegisterOrGet(LogLevel minLevel)
   at A.O.a()
   ```
   
   Here are the settings in the publish profile being used:
   ```
  <Configuration>Pro</Configuration>
  <Platform>Any CPU</Platform>
  <PublishDir>publish\windows\pro</PublishDir>
  <PublishProtocol>FileSystem</PublishProtocol>
  <_TargetId>Folder</_TargetId>
  <TargetFramework>net6.0</TargetFramework>
  <RuntimeIdentifier>win-x86</RuntimeIdentifier>
  <SelfContained>true</SelfContained>
  <IncludeNativeLibrariesForSelfExtract>true</IncludeNativeLibrariesForSelfExtract>
```
    
    I'm hoping you might have any idea of what we're doing wrong, or if you had any suggestions we might try.
    
    Thanks!

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for asking! I'll create an issue from this discussion as it looks like a bug.
